ophyd/device.py

```diff
@@ -157,8 +157,10 @@ def make_docstring(self, parent_class):
         return '\n'.join(doc)
 
     def __repr__(self):
-        kw_str = ', '.join('{}={!r}'.format(k, v)
-                           for k, v in self.kwargs.items())
+        for k, v in self.kwargs.items():
```

Comment:
This logic isn't quite right, and that is the source of Travis failure. In this implementation, `kw_str` gets re-defined on every pass through the loop.

I think it would be sufficient to go back to the original implementation and add `if k not in ['read_attrs','configuration_attrs']` after `items()`. That is, this kind of construction:

```python
# Add the even numbers from 0 to 9.
sum(i for i in range(10) if i % 2 == 0)
```

Reply:
I made this change as I think it is a neater way of solving the problem, although I am not sure why this would give the failure error that Travis gave. Anyway lets see if this solves the issue.

Reply:
I think the approach that triggered the error kept re-defining `kw_str` each time, blowing away contents from the previous iteration of the loop. The do this without a list comprehension, you'd have to define some sort of accumulator variable _before_ the loop. I think that's the approach you were going for here, which would also be fine. :- )

Reply:
thanks @danielballan for the explanation, I now realize why what I did didn't work. Still not sure how that generated the error message travis returned, but I guess I am missing something in how/when this requires importing of ophyd.utils.paths or datetime which is the command that travis complained about. Anyway not important now, it is all good.

Reply:
That ophyd.paths failure might have been a red herring, a random hiccup. The error that tipped me off to the `kw_str` issue was a `NameError`.

Reply:
thanks @danielballan, I think I took a wrong turn at the ophyd.paths failure and got stuck in the mud at the end !-). I didn't even notice the NameError until you mentioned it now and I took a look.